app/services/ingest_iati_activities.rb

```diff
@@ -210,8 +210,15 @@ def call
   end
 
   private def add_identifiers(legacy_activity:, new_activity:)
-    new_activity.identifier = SecureRandom.uuid
     new_activity.previous_identifier = legacy_activity.elements.detect { |element| element.name.eql?("iati-identifier") }.children.text
+    new_activity.identifier = infer_internal_identifier(old_identifier: new_activity.previous_identifier.dup)
```

Comment:
Could you help me to understand why the `dup` is required here?

Reply:
I believe I needed to `dup` because when I passed this value in: `new_activity.previous_identifier` the method then made changes to it which worked but changed the original object too which was not what I wanted. The value of `new_activity.previous_identifier` was mutated based on the behaviour of the infer_internal_identifier method.

I think I could have written it a different way to assign it to a new piece of memory but it's slightly more verbose so I skipped it:

```
previous_identifier = new_activity.previous_identifier
infer_internal_identifier(previous_identifier)
```